### ISSUE TITLE
Lowering the batch size

### ISSUE THREAD

**Issue by VCoklat** [serengil/deepface#18]
Hello.. i've tried to running your code in jetson nano and i've met error like this.

![image](https://user-images.githubusercontent.com/12172720/79593523-c9484e00-8105-11ea-8336-bf221ea70146.png)

I've tried to trying lowering the batch size but I think i do it wrong. I still got the same error. Can you help me? Thankyou

**Comment by serengil** [comment]
Could you add the following block in the very first lines of your program?

```
import os
os.environ["CUDA_VISIBLE_DEVICES"]="0"
```

**Comment by VCoklat** [comment]
Thankyou for your reply. I've added it before. The program can run but it's very slow and lagging because it's using the CPU.

**Comment by serengil** [comment]
You need very complex models and your gpu memory is not enough to allocate those models. This is the root cause of the problem. You can either increase the memory of gpu or use less complex models. The second choice will decrease the accuracy dramatically. 

BTW, what is the memory size of your gpu? could you share the screen when running the command "watch nvidia-smi".

**Comment by serengil** [comment]
The both verify and analyze functions return this error in jetson?

**Comment by serengil** [comment]
Could you set allow_growth option true as illustrated below before importing deepface? If this solves, then I put this block in the framework code.

```
import tensorflow as tf
import keras

config = tf.ConfigProto()
config.gpu_options.allow_growth = True
session = tf.Session(config=config)
keras.backend.set_session(session)
```

**Comment by serengil** [comment]
In deepface 0.0.20, I fixed the greedy approach of tensorflow. Could you upgrade the framework and test again?

**Comment by serengil** [comment]
You can re-open this issue if it is not solved in 0.0.20.

**Comment by VCoklat** [comment]
Hello thankyou for your reply, and i'm sorry i'm late. 
i've tried set allow_growth option true and update to 0.0.20 but the error still same.
I've tried "watch nvidia-smi" but it's not supported in nvidia jetson nano

**Comment by VCoklat** [comment]
@serengil hello, do you already have a solution for this problem?

**Comment by serengil** [comment]
I added a new function. This should be called before calling verify/analyze function.

`DeepFace.allocateMemory()`

However, this will cause to run on cpu if watch nvidia-smi command is not recognized. What is the memory size of your device?

**Comment by VCoklat** [comment]
@serengil thankyou for your reply. So we can't use GPU for this problem. Jetson nano has 4 GB of main memory

**Comment by serengil** [comment]
unfortunately, dependent models require 10 GB memory.

**Comment by VCoklat** [comment]
@serengil ah.. OK thankyou very much for the information
